doc/configuration-flow.md

```diff
@@ -0,0 +1,54 @@
+# Overview of the configuration flow in Istio Manager
+
+Istio configuration consists of configuration objects. Each object has kind
+(for example, a routing rule), name and namespace uniquely identifying the
+object.  Istio Manager provides a REST-based interface for manipulating objects
+by their keys. Objects can be listed by their kind and/or namespace, retrieved
+by the object keys, deleted, created (using "POST" operation), and updated
+(using "PUT" operation).
+
+The configuration registry also provides a caching controller interface that
+also allows watching for changes to the configuration store. For example, a
+function can subscribe to receive object creation events for objects of a
+particular kind. The controller streams the events to the subscribers, and also
+maintains a local cache view of the entire store.
+
+The flow of configuration starts at the operator, continues to the
+platform-specific persistent store, and then triggers a notification event at
+the remote proxy component.
+
+## User input
+
+Istio Manager provides a command line interface called `istioctl` that exposes
+the REST-based configuration store interface.  The tool validates that its YAML
+input satisfies the schema for the registered Istio configuration kind, and
+stores the validated input in the store.
+
```

Comment:
This is good. It might need some massaging, in terms of kind->type, etc. But the watering down of the message can be done later or in another doc that is for public consumption while we keep this for internal use (design doc).